File: taekyoung/20260109/py/arrow.py
best_score = 0
best = [0 for _ in range(11)]

def solution(n, info):
    global best, best_score
    answer = []
    
    for i in range(len(info)):
        if info[i] > 0:
            best_score -= 10 - i

    print(best_score)
            
    make_score(n, 0, best_score, info, [0 for _ in range(11)])
    print(best)

    if best_score <= 0:
        return [-1]
        
    return best

def make_score(n, k, cur_score, info, arrow):
    global best, best_score
    if n == 0 or k == 10:
        arrow[k] = n

        if cur_score > best_score:
            best_score = cur_score
            best = arrow
            return
        elif best_score > cur_score:
            return
        else:
            for i in range(10, -1, -1):
                if arrow[i] > best[i]:
                    best_score = cur_score
                    best = arrow
                    return
                elif best[i] > arrow[i]:
                    return
            return
    else:
        if n > info[k]:
            new_n = n - (info[k] + 1)
            new_arr = arrow[:]
            new_arr[k] = info[k] + 1
            new_score = 0
            if info[k] > 0:
                new_score = cur_score + (10 - k) * 2
            else:
                new_score = cur_score + 10 - k
            
            make_score(new_n, k + 1, new_score, info, new_arr)
        make_score(n, k + 1, cur_score, info, arrow)

File: taekyoung/20260109/cpp/arrow.cpp
#include <string>
#include <vector>

using namespace std;
vector<int> make_score(vector<int> info, vector<int> scores, vector<int> &best, int n, int k, int diff, int &best_diff);

vector<int> solution(int n, vector<int> info) {
    
    int len = info.size();
    vector<int> temp(len);
    int score = 0;
    
    for (int i = 0; i < len; i ++)
    {
        score -= info[i] ? 10 - i : 0;
    }
    
    vector<int> answer = make_score(info, temp, temp, n, 0, score, score);

    if (score <= 0) return {-1};
    return answer;
}

vector<int> make_score(vector<int> info, vector<int> scores, vector<int> &best, int n, int k, int diff, int &best_diff)
{
    if (n == 0 || k == 10)
    {       
        scores[k] = n;
        if (diff > best_diff)
        {
            best = scores;
            best_diff = diff;
            return scores;
        }
        else if (best_diff > diff)
        {
            return best;
        }
        else
        {
            for(int i = 10; i >= 0; i --)
            {
                if (scores[i] > best[i])
                {
                    best = scores;
                    best_diff = diff;
                    return scores;
                }
                else if (scores[i] < best[i]) return best;
            }
            return best;
        }
    }
    else
    {
        if (n > info[k])
        {
            vector<int> new_score = scores;
            new_score[k] = info[k] + 1;
            int new_diff = info[k] ? diff + (10 - k) * 2 : diff + (10 - k);
            make_score(info, new_score, best, n - (info[k] + 1), k + 1, new_diff, best_diff);
        }
        make_score(info, scores, best, n, k + 1, diff, best_diff);
    }
    return best;
}